### PR TITLE
 Bug 1924641: Remove error message for "missing storage class"

### PR DESCRIPTION
**Pull request by rexagod**
Mentioning here that I was able to replicate this issue only a couple
of times (as is mentioned in the BZ issue body), so this patch fix
addresses some parts of the codebase that I inferred to be the cause
of this issue from that observation.

It'd be nice if someone (who can replicate the issue successfully) can
apply this patch and provide their feedback on it.

***
To apply this, do `curl https://github.com/openshift/console/commit/f12891f445c73f0041c397813e9dedbeb36b8fe8.patch | git apply`.

### CODE REVIEW

**Comment by rexagod** [comment]
Pasting this in for reference: https://bugzilla.redhat.com/show_bug.cgi?id=1931543

**Comment by rexagod** [comment]
To add to the aforementioned information, I can confirm that this patch basically checks when the provSC gets filtered in the storage classes dropdown, for the noProv storage class, and invokes the noProv [detection and updation process](https://github.com/rexagod/console/blob/f12891f445c73f0041c397813e9dedbeb36b8fe8/frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install.tsx#L62).

**Comment by rexagod** [comment]
PS. To apply this, do `curl https://github.com/openshift/console/commit/f12891f445c73f0041c397813e9dedbeb36b8fe8.patch | git apply`.

**Review by afreen23 — CHANGES_REQUESTED** [review]
> Mentioning here that I was able to replicate this issue only a couple
> of times (as is mentioned in the BZ issue body), so this patch fix
> addresses some parts of the codebase that I inferred to be the cause
> of this issue from that observation.
> 
> It'd be nice if someone (who can replicate the issue successfully) can
> apply this patch and provide their feedback on it.
> 
> To apply this, do `curl https://github.com/openshift/console/commit/f12891f445c73f0041c397813e9dedbeb36b8fe8.patch | git apply`.

The issue is due to the fact that when we create storage class, we dont fetch results again to update that.
On a new mount this is not the case and storage class is found and hence the correct display.
We need to fix this issue by fetching storage class in wizard and looking if a no  provsc is there, that means moving no Prov sc to wizard component.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/create-sc/wizard-pages/create-local-volume-set.tsx [hunk 1]
```diff
@@ -167,7 +177,14 @@ const ConfirmationModal = ({ state, dispatch, setInProgress, setErrorMessage, ns
...
             variant="small"
             actions={[
-              <Button key="confirm" variant="primary" onClick={makeLVSCall}>
+              <Button
+                key="confirm"
+                variant="primary"
+                onClick={() => {
+                  makeLVSCall();
+                  sendDetectionReq(true);
+                }}
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/create-sc/wizard-pages/create-local-volume-set.tsx:186]: move it out toa function maybe call it `onCreate`

**Comment by rexagod** [comment]
Here a visual of the changes.

https://user-images.githubusercontent.com/33557095/110671526-9d89f700-81f4-11eb-9b66-c78dba97f4a2.mp4

**Review by afreen23 — CHANGES_REQUESTED** [review]
Run `yarn i18n` and update texts

**Comment by rexagod** [comment]
Updating the i18n files.

**Comment by rexagod** [comment]
Found a bug, updating this ASAP.

**Comment by rexagod** [comment]
I've removed hasNoProvSC and setHasNoProvSC entirely from the codebase since we aren't using those anywhere.

**Comment by rexagod** [comment]
Fixing conflicts.

**Review by afreen23 — APPROVED** [review]
/lgtm

**Pull request merged**